client/components/editor/teaching-elements/Assessment/ImageMultipleChoice.vue

```diff
@@ -0,0 +1,319 @@
+      </li>
+    </ul>
+  </div>
+</template>
+
+<script>
+import cloneDeep from 'lodash/cloneDeep';
+import range from 'lodash/range';
+import { blobToBase64String } from 'blob-util';
+
+const maxImageDimension = 256;
+
+const formAlert = (message) => {
+  return {
+    type: 'alert-danger',
+    text: message
+  };
+};
+
+export default {
+  props: {
+    assessment: Object,
+    isEditing: Boolean,
+    errors: Array,
+    fileInputTypes: { type: Array, default: () => ['image/jpeg', 'image/png'] }
+  },
+  data() {
+    return {
+      fieldErrors: []
+    };
+  },
+  computed: {
+    answers() {
+      return this.assessment.answers;
+    },
+    correct() {
+      return this.assessment.correct;
+    },
+    feedback() {
+      return this.assessment.feedback;
+    },
+    disabled() {
+      return !this.isEditing;
+    },
+    imageInputType() {
+      return this.fileInputTypes.join(',');
+    }
+  },
+  methods: {
+    update(data) {
+      this.$emit('update', data);
+    },
+    toggleAnswer(index) {
+      let correct = cloneDeep(this.correct);
+      const position = correct.indexOf(index);
+
+      if (position < 0) {
+        correct.push(index);
+      } else {
+        correct.splice(position, 1);
+      }
+
+      this.update({ correct });
+    },
+    updateAnswer(evt) {
+      let answers = cloneDeep(this.answers);
+      const { target: input } = evt;
+      const { dataset, files = [] } = input;
+      const imageFile = files[0];
+
+      return new Promise((resolve, reject) => {
+        this.resetErrorMessages(dataset.index);
+        if (imageFile.size > 100000) {
+          // eslint-disable-next-line prefer-promise-reject-errors
+          return reject('The image is too large. Maximum allowed image size: 100 kB');
+        }
+        return resolve(imageFile);
+      })
+        .then(() => blobToBase64String(imageFile))
+        .then(base64String => {
+          answers[dataset.index] = base64String;
+          return this.getImageDimensions(base64String);
+        })
+        .then(({ width, height }) => {
+          if (width > maxImageDimension || height > maxImageDimension) {
+            // eslint-disable-next-line prefer-promise-reject-errors
+            return Promise.reject(
+              `Incorrect image width and/or height. Maximum allowed image dimensions:
+               ${maxImageDimension}x${maxImageDimension}`
+            );
+          }
+        })
+        .catch(errorMsg => {
+          this.addErrorMessage(dataset.index, errorMsg);
+          answers[dataset.index] = '';
+        })
+        .then(() => {
+          this.update({ answers });
+        });
+    },
+    getImageDimensions(base64String) {
+      return new Promise((resolve, reject) => {
+        let i = document.createElement('img');
+        i.onload = () => {
+          resolve({ width: i.width, height: i.height });
+        };
+        i.src = this.getSrc(base64String);
+      });
```

Comment:
Nice but extract this to helper function and use inline return inside `onload` callback.